### ISSUE TITLE
Label.FormattedText does not support FontAttributes on Android, crashes on Windows

### ISSUE THREAD

**Issue by Dbquity** [dotnet/maui#5576]
### Description

Changing the FormattedText on a Label crashes on Windows (IMauiContext not found) and ignores spans with font attributes on Android.

### Steps to Reproduce

1. Create a new MAUI App

2. replace the MainPage.xaml.cs with this (you will likely need to correct the namespace):
```csharp
namespace MauiApp7;
using static FontAttributes;
public partial class MainPage : ContentPage {
    static readonly FontAttributes[] attributes = new[] { None, Italic, None, Bold, None };
    int count = 0;
    public MainPage() => InitializeComponent();
    private void OnCounterClicked(object sender, EventArgs e) {
        if (++count < 5)
            CounterLabel.Text = $"Current count: {count}";
        else
            CounterLabel.FormattedText = new FormattedString {
                Spans = {
                    new Span { Text = "Current ", FontAttributes = attributes[count % 5] },
                    new Span { Text = "count: ", FontAttributes = attributes[(count + 1) % 5] },
                    new Span { Text = count.ToString(), FontAttributes = attributes[(count + 2) % 5] }
                }
            };
    }
}
```

3. run on Android, click more than five times on [Click me] and observe that only the spans with ```FontAttributes.None``` show:
![image](https://user-images.githubusercontent.com/25459675/160142486-32a993c3-7efd-4018-b3c6-dc37ebbbf998.png)

4. run on Windows and observe the crash at the fifth click on [Click me]:
```
System.InvalidOperationException
  HResult=0x80131509
  Message=IMauiContext not found.
  Source=Microsoft.Maui.Controls
  StackTrace:
   at Microsoft.Maui.Controls.ViewExtensions.RequireMauiContext(Element element, Boolean fallbackToAppMauiContext)
   at Microsoft.Maui.Controls.Platform.FormattedStringExtensions.UpdateInlines(TextBlock textBlock, FormattedString formattedString, Double defaultLineHeight, TextAlignment defaultHorizontalAlignment, Nullable`1 defaultFont, Color defaultColor, TextTransform defaultTextTransform)
   at Microsoft.Maui.Controls.Platform.FormattedStringExtensions.UpdateInlines(TextBlock textBlock, Label label)
   at Microsoft.Maui.Controls.Platform.TextBlockExtensions.UpdateText(TextBlock platformControl, Label label)
   at Microsoft.Maui.Controls.Label.MapText(LabelHandler handler, Label label)
   at Microsoft.Maui.Controls.Element.OnPropertyChanged(String propertyName)
   at Microsoft.Maui.Controls.BindableObject.SetValueActual(BindableProperty property, BindablePropertyContext context, Object value, Boolean currentlyApplying, SetValueFlags attributes, Boolean silent)
   at Microsoft.Maui.Controls.BindableObject.SetValueCore(BindableProperty property, Object value, SetValueFlags attributes, SetValuePrivateFlags privateAttributes)
   at Microsoft.Maui.Controls.BindableObject.SetValue(BindableProperty property, Object value, Boolean fromStyle, Boolean checkAccess)
   at Microsoft.Maui.Controls.BindableObject.SetValue(BindableProperty property, Object value)
   at Microsoft.Maui.Controls.Label.set_FormattedText(FormattedString value)
   at MauiApp7.MainPage.OnCounterClicked(Object sender, EventArgs e) in C:\Code\Maui\MauiApp7\MainPage.xaml.cs:line 11
   at Microsoft.Maui.Controls.Button.Microsoft.Maui.Controls.Internals.IButtonElement.PropagateUpClicked()
   at Microsoft.Maui.Controls.ButtonElement.ElementClicked(VisualElement visualElement, IButtonElement ButtonElementManager)
   at Microsoft.Maui.Controls.Button.SendClicked()
   at Microsoft.Maui.Controls.Button.Microsoft.Maui.IButton.Clicked()
   at Microsoft.Maui.Handlers.ButtonHandler.OnClick(Object sender, RoutedEventArgs e)
   at WinRT._EventSource_global__Microsoft_UI_Xaml_RoutedEventHandler.EventState.<GetEventInvoke>b__1_0(Object sender, RoutedEventArgs e)
   at ABI.Microsoft.UI.Xaml.RoutedEventHandler.<>c__DisplayClass10_0.<Do_Abi_Invoke>b__0(RoutedEventHandler invoke)
   at ABI.Microsoft.UI.Xaml.RoutedEventHandler.Do_Abi_Invoke(IntPtr thisPtr, IntPtr sender, IntPtr e)

```

### Version with bug

Preview 14 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android, Windows, I was *not* able test on other platforms

### Affected platform versions

Windows 10, Android emulator Pixel 5 - API 30

### Did you find any workaround?

No

**Comment by Redth** [comment]
Ok seems as though when set in XAML it works (as per the sample in our repo that didn't catch this) however setting/changing it from code is throwing here on windows too.

**Comment by Dbquity** [comment]
Impressively fast response!
Just double-checking: does the fix solve the Android issue as well as the crash on Windows?
Otherwise, I'll be happy to open a new issue specifically for Android, if that is helpful :-)

**Comment by Redth** [comment]
I thought Android was working when I tested it but i'll reopen this just in case and validate again.

**Comment by JohnHDev** [comment]
The IMauiContext issue was already reported, @Redth csn you link the issues pls.

**Comment by Redth** [comment]
Related to #5350 and fixed by #5586
